### PR TITLE
Retrieve transaction by ID - Closes #238

### LINKED ISSUE #238: Retrieve transaction by ID
### Description

Make Lisk Service possible to retrieve transactions by ID. 

Add `get.transactions` API route to the Gateway.

Endpoint: `/api/v2/transactions`

### Motivation

- The basic retrieval by ID is essential for the further development

### Acceptance Criteria

- Transactions can be retrieved by ID
- Gateway endpoint is created

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #238 

### How was it solved?
- [x]  HTTP /api/v2/transactions works
- [x]  RPC get.transactions works
- [x] Transaction retrieval by id and ids
- [x]  Response map properly
- [x] Add moduleID/moduleName and assetID/assetName to in-memory during startup

### How was it tested?
Local

### CODE REVIEW

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
The implementation mostly looks good to me apart from the minor suggestions (mostly good to have) that I've added.

File: services/core/shared/core/compat/sdk_v5/coreApi.js [hunk 1]
```diff
@@ -26,6 +26,22 @@ const getNetworkStatus = async () => {
...
+    let transaction;
+    let transactions;
+
+    if (params.id) {
+        transaction = await apiClient.transaction.get(params.id);
+    } else if (params.ids) {
+        transactions = await apiClient._channel.invoke('app:getTransactionsByIDs', { ids: params.ids });
+    }
+
+    if (transactions) transactions = transactions.map(blk => apiClient.transaction.decode(blk));
```
[the remark this answers is not in the document]
Comment (nagdahimanshu) [reply services/core/shared/core/compat/sdk_v5/coreApi.js]: Done!

File: services/core/shared/core/compat/sdk_v5/transactions.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	// timestamp logic for transactions
+	// transactions.data = await BluebirdPromise.map(
+	// 	transactions.data,
+	// 	async transaction => {
+	// 		if (!transaction.timestamp) {
+	// 			const txBlock = (await coreApi.getBlocks({ height: transaction.height })).data[0];
+	// 			transaction.timestamp = txBlock.timestamp;
+	// 		}
+	// 		transaction.unixTimestamp = await getUnixTime(transaction.timestamp);
+	// 		return transaction;
```
[the remark this answers is not in the document]
Comment (nagdahimanshu) [reply services/core/shared/core/compat/sdk_v5/transactions.js:54]: Done!

File: services/gateway/apis/http-version2/methods/transactions.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+		height: { optional: true, type: 'string', min: 1 },
+		limit: { optional: true, type: 'number', min: 1, max: 100 },
+		offset: { optional: true, type: 'number', min: 0 },
+		sort: {
+			optional: true,
+			type: 'string',
+			min: 1,
+			enum: [
+				'amount:asc', 'amount:desc', 'timestamp:asc', 'timestamp:desc',
+			],
```
[the remark this answers is not in the document]
Comment (nagdahimanshu) [reply services/gateway/apis/http-version2/methods/transactions.js]: Done!

File: services/gateway/apis/http-version2/methods/transactions.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+module.exports = {
+	version: '2.0',
+	swaggerApiPath: '/transactions',
+	rpcMethod: 'get.transactions',
+	tags: ['Transactions'],
+	params: {
+		id: { optional: true, type: 'string', min: 1 },
+		operationId: { optional: true, type: 'string', min: 1 },
+		operationName: { optional: true, type: 'string', min: 1 },
+		address: { optional: true, type: 'string', min: 1 },
```
[the remark this answers is not in the document]
Comment (nagdahimanshu) [reply services/gateway/apis/http-version2/methods/transactions.js]: Done!

File: services/gateway/apis/http-version2/methods/transactions.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+const transactionsSource = require('../../../sources/version2/transactions');
+const envelope = require('../../../sources/version2/mappings/stdEnvelope');
+
+module.exports = {
+	version: '2.0',
+	swaggerApiPath: '/transactions',
+	rpcMethod: 'get.transactions',
+	tags: ['Transactions'],
+	params: {
+		id: { optional: true, type: 'string', min: 1 },
```
[the remark this answers is not in the document]
Comment (nagdahimanshu) [reply services/gateway/apis/http-version2/methods/transactions.js]: Done!

File: services/gateway/apis/http-version2/methods/transactions.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+		operationId: { optional: true, type: 'string', min: 1 },
```
[the remark this answers is not in the document]
Comment (nagdahimanshu) [reply services/gateway/apis/http-version2/methods/transactions.js]: Done!

File: services/gateway/apis/http-version2/methods/transactions.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+	swaggerApiPath: '/transactions',
+	rpcMethod: 'get.transactions',
+	tags: ['Transactions'],
+	params: {
+		id: { optional: true, type: 'string', min: 1, max: 64 },
+		operationId: { optional: true, type: 'string', min: 1, max: 3 },
+		operationName: { optional: true, type: 'string', min: 1 },
+		address: { optional: true, type: 'string', min: 38, max: 41 },
+		sender: { optional: true, type: 'string', min: 1 },
+		recipient: { optional: true, type: 'string', min: 1 },
```
Comment (sameersubudhi) [inline services/gateway/apis/http-version2/methods/transactions.js:30]:
```suggestion
		sender: { optional: true, type: 'string', min: 1, max: 64 },
		recipient: { optional: true, type: 'string', min: 1, max: 64 },
```

**Review by sameersubudhi — APPROVED** [review]
Looks good overall! Thank you 👍

File: services/gateway/apis/http-version2/methods/transactions.js (hunk 7 above)
Comment (nagdahimanshu) [reply services/gateway/apis/http-version2/methods/transactions.js]: Done!

File: services/core/shared/core/compat/sdk_v5/network.js [hunk 8]
```diff
@@ -14,23 +14,13 @@
...
 	const { offset } = status.data.genesisConfig.rewards;
 	const { distance } = status.data.genesisConfig.rewards;
-	status.data.operations = [];
-	status.data.registeredModules.forEach(liskModule => {
-		if (liskModule.transactionAssets.length) {
-			status.data.operations = status.data.operations
-				.concat(
-					liskModule.transactionAssets.map(asset => {
-						const id = String(liskModule.id).concat(':').concat(asset.id);
-						const name = liskModule.name.concat(':').concat(asset.name);
```
Comment (MichalTuleja) [inline services/core/shared/core/compat/sdk_v5/network.js:29]: Please handle the name as optional property

**Review by MichalTuleja — CHANGES_REQUESTED** [review]

File: services/core/shared/core/compat/sdk_v5/network.js (hunk 8 above)
Comment (nagdahimanshu) [reply services/core/shared/core/compat/sdk_v5/network.js:29]: Done!

**Review by MichalTuleja — APPROVED** [review]

**Pull request merged**